common/gsutil.py

```diff
@@ -64,19 +64,17 @@ def rsync(  # pylint: disable=too-many-arguments
         destination,
         delete=True,
         recursive=True,
-        gsutil_options=None,
         options=None,
         **kwargs):
     """Does gsutil rsync from |source| to |destination| using sane defaults that
-    can be overriden. Prepends any |gsutil_options| before the rsync subcommand
-    if provided."""
-    command = [] if gsutil_options is None else gsutil_options
+    can be overriden. Prepends any |options['gsutil']| before the rsync
+    subcommand if provided."""
+    command = [] if (options is None) or (
```

Comment:
Break up this if into multiple lines please.

Reply:
I think having a `gsutil_options` is cleaner than having a `gsutil` key in `options`

Reply:
I think the dictionary way is cleaner and easier for scaling other file storage.
For a general file storage, it should not have a function parameter called `gsutil_options`. However, it can say, if you switch to `gsutil`, then check whether it has additional options to apply.

Reply:
Or, for `gsutil.py`, we can keep `gsutil_options`, but for `filestore_utils.py`, we'd better not support that. Otherwise, when new file storage supported, we need to add another `fs1_options`, `fs2_options` in the function signature. I think that is not what we want.

Reply:
@zchcai, that's a good point. However, at this point we should follow the principle of YAGNI and do the simplest thing, which is to just have a `gsutil_options` for now. Later we might find a nicer ways to generalize this interface.

Reply:
From your previous comment, https://github.com/google/fuzzbench/pull/394#discussion_r434228430, I think we should remove `gsutil_options` from `filestore_utils.py`.
For a generic interface, `gsutil_options` is not suitable.
And, when I add `local_utils` later, I also want to convert it to support local-specific options, and not adding another `local_options`.
Better scaling ability doesn't mean redundant. Additionally, currently, this is a very small change, rather than you thought. This is simplest change comparing to upcoming local support. Thanks.

Reply:
Sorry about the confusion, I should have been more clear. I was hoping that we could get rid passing gsutil specific stuff completely, e.g., as suggested in https://github.com/zchcai/fuzzbench/pull/7#discussion_r434297063.

With this PR we don't really get rid of it, we essentially just "hide it" behind the `options` dict parameter. While this does make the interface more generic, it also make it a bit more difficult to use and understand. As mentioned on the above comment, at this point it's probably better to prioritize the latter. Wdyt?

Reply:
> Break up this if into multiple lines please.

Done. @jonathanmetzman

Reply:
> I was hoping that we could get rid passing gsutil specific stuff completely

I see. That is another story. :-) Probably, we cannot assume every `rsync` will share a same optional option.

Reply:
I'm not entirely sure where this ended up, but I'd like to leave gsutil_options here. 
And I agree with Laszlo about YAGNI. Leave it as gsutil_options instead of hiding it please.